### ISSUE TITLE
Always have Open With option for any file

### ISSUE THREAD

**Issue by vavavr00m** [moudey/Shell#290]
I need to set batchfiles to open with CMD. Non-Windows Store Apps aren't selectable. I prefer not to reset all associations. How am I able to have file context menu to always have Open With with Shell to associate a file type with any program I want?

![image](https://github.com/moudey/Shell/assets/4675132/45057199-33d4-462a-bf06-cf3572276693)

![image](https://github.com/moudey/Shell/assets/4675132/61725b56-e4e8-471e-bc2a-fa0c1fc1222b)

**Comment by moudey** [comment]
```
item(title="open batfile" find=".bat|.cmd" cmd='sel.path')
```
Pass arguments
```
item(title="open batfile" find=".bat|.cmd" cmd='sel.path' args="arguments")
```

**Comment by vavavr00m** [comment]
@moudey Thanks. How/Where do I add that line? I can see there's .bat .cmd there..

Additionally, is that for ..
>always have Open With with Shell to associate a file type with any program

or for 

>restored the batchfile association using [this](https://superuser.com/a/54280) method but how about with Shell

**Comment by moudey** [comment]
Add it to the dynamic section of the 'shell.nss' file.
Save changes, press <kbd>CTRL</kbd> + <kbd>RIGHT-CLICK</kbd> to reload settings.

**Comment by vavavr00m** [comment]
Added and reloaded. Open batfile appeared but can you tell me about what it does exactly? Clicking it doesn't do anything. Batchfiles & Cmdfiles are still associated with TakeCommand and I am unable to find the "Open With" dialog with any .bat or .cmd file.

**Comment by moudey** [comment]
Excuse me. Do you need to open the "Open With" dialog box?

**Comment by vavavr00m** [comment]
Yes, that's what's needed because something got messed up after uninstalling TakeCommand.

**Comment by moudey** [comment]
With the latest debug [build](https://nilesoft.org/download/shell/debug.zip) the "openas" in the "verb" property has been corrected, as it shows the Open With dialog

```
item(title='test Open with' verb="openas" cmd=sel.path)
```

**Comment by vavavr00m** [comment]
Hello. I uninstalled 1.8.1 and registered the debug build. Two issues with the debug build:

1. The context menu is now taking up a space. I was using the installer or did I get it from winget... regardless, it looked ok.
2. no dynamic section in shell.nss - does it have to be added somewhere? Not seeing any test Open with in the context menu.

**Comment by moudey** [comment]
> The context menu is now taking up a space. I was using the installer or did I get it from winget... regardless, it looked ok.

There is only a change in the font size, you can modify or delete it.

> no dynamic section in shell.nss - does it have to be added somewhere? Not seeing the 'Open with' option for batfiles and cmdfiles.

There are modifications in the sections where the variables, static and dynamic sections have been removed

**Comment by vavavr00m** [comment]
> There is only a change in the font size, you can modify or delete it.

I will try to find this...

> There are modifications in the sections where the variables, static and dynamic sections have been removed

This makes no sense to me and did not answer my questions directly. Can the explanation/instruction be further detailed?

**Comment by moudey** [comment]
You will find the option to modify the font in the "imports/theme.nss" file.
```
font.size=20
```

Add this code at the end of the "shell.nss" file
```
item(title='test Open with' verb="openas" cmd=sel.path)
```

**Comment by vavavr00m** [comment]
> You will find the option to modify the font in the "imports/theme.nss" file.
> 
> ```
> font.size=20
> ```

Gotcha. Thanks.

> Add this code at the end of the "shell.nss" file
> 
> ```
> item(title='test Open with' verb="openas" cmd=sel.path)
> ```

The 'test Open with' still doesn't do anything with .bat or .cmd. Am I doing something wrong?

![image](https://github.com/moudey/Shell/assets/4675132/626de2a7-2a0f-4a6c-a11d-0caf4f9121ff)


![image](https://github.com/moudey/Shell/assets/4675132/cd271af6-7498-4a1a-944f-da67a5d909ed)
